app/ante/handlers.go

```diff
@@ -0,0 +1,103 @@
+		evmante.NewEthEmitEventDecorator(options.EvmKeeper),
+	)
+}
+
+func newCosmosAnteHandler(options HandlerOptions) sdk.AnteHandler {
+	return sdk.ChainAnteDecorators(
+		cosmosDecorators(
+			options,
+			options.ExtensionOptionChecker, // make sure there are no extension options
+			ante.NewSigVerificationDecorator(options.AccountKeeper, options.SignModeHandler), // Use modern signature verification
+		)...,
+	)
+}
+
+// Deprecated: NewLegacyCosmosAnteHandlerEip712 creates an AnteHandler to process legacy EIP-712
+// transactions, as defined by the presence of an ExtensionOptionsWeb3Tx extension.
+func newLegacyCosmosAnteHandlerEip712(options HandlerOptions) sdk.AnteHandler {
+	return sdk.ChainAnteDecorators(
+		cosmosDecorators(
+			options,
+			func(c *codectypes.Any) bool { return true },                                                       // allows any extension, this is consistent with the evmos behavior
+			cosmosante.NewLegacyEip712SigVerificationDecorator(options.AccountKeeper, options.SignModeHandler), // Use old signature verification: uses EIP instead of the cosmos signature validator
+		)...,
+	)
+}
+
+func cosmosDecorators(options HandlerOptions, extensionChecker authante.ExtensionOptionChecker, sigChecker sdk.AnteDecorator) []sdk.AnteDecorator {
+	sigGasConsumer := options.SigGasConsumer
+	if sigGasConsumer == nil {
+		sigGasConsumer = authante.DefaultSigVerificationGasConsumer
+	}
+	return []sdk.AnteDecorator{
+		cosmosante.NewRejectMessagesDecorator(
+			[]string{
+				sdk.MsgTypeURL(&evmtypes.MsgEthereumTx{}),
+			},
+		),
+		cosmosante.NewAuthzLimiterDecorator( // disable the Msg types that cannot be included on an authz.MsgExec msgs field
+			sdk.MsgTypeURL(&evmtypes.MsgEthereumTx{}),
+			sdk.MsgTypeURL(&sdkvestingtypes.MsgCreateVestingAccount{}),
+			sdk.MsgTypeURL(&sdkvestingtypes.MsgCreatePermanentLockedAccount{}),
+			sdk.MsgTypeURL(&sdkvestingtypes.MsgCreatePeriodicVestingAccount{}),
+		),
+		ante.NewSetUpContextDecorator(),
+		ante.NewExtensionOptionsDecorator(extensionChecker),
+		ante.NewValidateBasicDecorator(),
+		ante.NewTxTimeoutHeightDecorator(),
+		NewPermissionedURLsDecorator(
+			func(ctx sdk.Context, accAddr sdk.AccAddress) bool {
+				return options.hasPermission(ctx, accAddr, vestingtypes.ModuleName)
+			}, []string{
+				sdk.MsgTypeURL(&vestingtypes.MsgCreateVestingAccount{}),
+				sdk.MsgTypeURL(&vestingtypes.MsgCreatePermanentLockedAccount{}),
+				sdk.MsgTypeURL(&vestingtypes.MsgCreatePeriodicVestingAccount{}),
+			}),
+		ante.NewValidateMemoDecorator(options.AccountKeeper),
+		cosmosante.NewMinGasPriceDecorator(options.FeeMarketKeeper, options.EvmKeeper),
+		ante.NewConsumeGasForTxSizeDecorator(options.AccountKeeper),
+		cosmosante.NewDeductFeeDecorator(options.AccountKeeper, options.BankKeeper, options.DistributionKeeper, options.FeegrantKeeper, options.StakingKeeper, options.TxFeeChecker),
+		cosmosante.NewVestingDelegationDecorator(options.AccountKeeper, options.StakingKeeper, options.Cdc),
```

Comment:
not sure what it is. we don't have it on the hub e.g

Reply:
hmm 
it's from evmos
https://github.com/dymensionxyz/evmos/blob/2dcb56724c6cd1610db205af5ee12d84b608828f/app/ante/handler_options.go#L133

Reply:
no reason to keep it

Reply:
ok done 👍